File: minDistTriplet.java
int minDistTriplet(int[] a, int[] b, int[] c) {
    int i = 0, j = 0, k = 0;
    int minimum = 999999;
    while(i < a.length && j < b.length && k < c.length){
        int result = Math.abs(a[i] - b[j]) + Math.abs(b[j] - c[k]) + Math.abs(c[k] - a[i]);
        minimum = Math.min(minimum, result);
        int small = Math.min(a[i], Math.min(b[j], c[k]));
        
        if (small == a[i])
            i++;
        if (small == b[j])
            j++;
        if (small == c[k])
            k++;
    }
    return minimum;
}

File: Codesignal and Geekforgeeks Combined/minDistTriplet.py
def minDistTriplet(a, b, c):
    i, j, k = 0, 0, 0
    minimum = 999999999
    
    while i < len(a) and j < len(b) and k < len(c):
        result = abs(a[i] - b[j]) + abs(b[j] - c[k]) + abs(c[k] - a[i])
        minimum = min(minimum, result)
        m = min(a[i], b[j], c[k])
        
        if m == a[i]:
            i += 1
        if m == b[j]:
            j += 1
        if m == c[k]:
            k += 1
    
    return minimum
